presto-main/src/main/java/io/prestosql/operator/DynamicFilterSourceOperator.java

```diff
@@ -233,26 +332,42 @@ public void finish()
             return;
         }
         finished = true;
+        ImmutableMap.Builder<DynamicFilterId, Domain> domainsBuilder = new ImmutableMap.Builder<>();
         if (valueSets == null) {
-            return; // the predicate became too large.
+            // the predicate became too large
+            if (minValues != null && maxValues != null) {
+                // create TupleDomain from min/max values
+                for (Integer channelIndex : minMaxChannels) {
+                    Type type = channels.get(channelIndex).type;
+                    Object min = readNativeValue(type, minValues[channelIndex], 0);
```

Comment:
what if `minValues[channelIndex]` is null? That would mean there were only nulls. You should return `TupleDomain.none` then

Reply:
Right, fixed this and added test